ha/programmers-1/같은숫자는싫어.js

```diff
@@ -1,3 +1,40 @@
-function solution(arr) {
-  // 풀이2
+function solution(s) {
+  let str = s.toUpperCase();
+  let count = 0;
+
+  for (let i = 0; i < s.length; i++) {
+    if (str[i] === "P") count++;
+    else if (str[i] == "Y") count--;
+  }
+  return count === 0 ? true : false;
 }
```

Comment:
```suggestion
function solution(s) {
  const upperStr = s.toUpperCase(); // 좀 더 명확한 변수명
  let count = 0;

  for (let i = 0; i < s.length; i++) {
    if (str[i] === "P") count++;
    else if (str[i] == "Y") count--;
  }
  return count === 0 ? true : false;
}
```
- 하연님 잘푸셨네요 👍 약간의 피드백을 드리자면 str이라는 변수보다는 upperStr이라는 변수가 좀 더 명확해보입니다.
- 그리고 비교할 때 ==(동등 연산자), === (일치 연산자)가 존재하는데 ==동등 연산자보다는 === 일치 연산자를 사용해주세요. 여러가지가 이유가 있는데 좀 더 정확하게 비교를 해준다는 특징이 있습니다.
- 그리고 기본적인 for문도 있지만 for...of문으로도 가능하다는 점 참고해주시면 좋을 것 같아요 아래는 for ... of문 예제입니다 🙇‍♂️

<br />

```js
// for ... of문 예제
  for (const el of upperStr) {
    if (el === "P") count++;
    else if (el === "Y") count--;
  }
```

Reply:
피드백 너무 감사합니다! 저도 처음에 접근이 어러워서 구글링으로 도움을 받았지만,,ㅎㅎ 연산자 개념 다시 정리해놓겠습니당!! for...of문은 스스로 써 본적이 없어서 생각을 못했네요!!!!! 우왁 한번 for...of 문으로도 다시 연습해보겠습니다 감사해요!!👍🏻👍🏻